src/util/sss_regexp.c

```diff
@@ -0,0 +1,144 @@
+/*
+    SSSD
+
+    Authors:
+        Tomas Halman <thalman@redhat.com>
+
+    Copyright (C) 2018 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "util/sss_regexp.h"
+#include <string.h>
+
+#define SSS_REGEXP_OVEC_SIZE 30
+#define SSS_REGEXP_ERR_MSG_SIZE 120 // 120 is recomended by pcre2 doc
+
+#ifdef HAVE_LIBPCRE2
+
+struct _sss_regexp_t {
+    pcre2_code *re;
+    unsigned char *error_msg;
+    pcre2_match_data *match_data;
+    char *matched_string;
+};
+
+#else /* HAVE_LIBPCRE2 */
+
+struct _sss_regexp_t {
+    pcre *re;
+    int ovector[SSS_REGEXP_OVEC_SIZE];
+    const char *matched_string;
+    const char *subject;
+};
+
+#endif /* HAVE_LIBPCRE2 */
+
+
+int sss_regexp_destroy (sss_regexp_t *self)
+{
+    if (! self) return -1;
+#ifdef HAVE_LIBPCRE2
+    if (self->re) pcre2_code_free(self->re);
+    if (self->error_msg) free (self->error_msg);
+    if (self->match_data) pcre2_match_data_free(self->match_data);
+    if (self->matched_string) pcre2_substring_free((PCRE2_UCHAR *)self->matched_string);
+#else /* HAVE_LIBPCRE2 */
+    if (self->re) pcre_free(self->re);
+    if (self->matched_string) pcre_free_substring(self->matched_string);
+#endif /* HAVE_LIBPCRE2 */
+    return 0;
+}
+
+
+sss_regexp_t *sss_regexp_new(
+    TALLOC_CTX *mem_ctx,
+    const char *pattern,
+    int options,
+    int *errorcodeptr,
+    const char **errormsgptr,
+    size_t *erroffset
```

Comment:
I must say I am not a big fan of this indentation style. We usually use:
```
int foo(int param1
           int param2)
```

Reply:
OK, apparently I cannot format my own examples well either :) but what I meant is that the first parameter is on the same line as the function name and the continuation line is indented to the same column as the first parameter.